src/App.tsx

```diff
@@ -1,26 +1,214 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, getTodos, patchTodos, postTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { Tabs } from './types/Tabs';
+import { Footer } from './components/Footer';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [input, setInput] = useState('');
+  const [formKey, setFormKey] = useState(0);
+  const [selectedTab, setSelectedTab] = useState(Tabs.All);
+  const [error, setError] = useState('');
+  const [isErrorShown, setIsErrorShown] = useState(false);
+  const [isLoading, setIsLoading] = useState<number[]>([]); // ids of the changed todos
```

Comment:
a good approach to handle loading status for different todos :+1: 
But now you store array, not boolean value here, so let's rename it to smth like `loadingTodoIds` - id describes better the stored value